java/code/src/com/redhat/rhn/common/util/test/DebVersionComparatorTest.java

```diff
@@ -0,0 +1,120 @@
+/**
+ * Copyright (c) 2020 SUSE LLC
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package com.redhat.rhn.common.util.test;
+
+import com.redhat.rhn.common.util.DebVersionComparator;
+
+import junit.framework.TestCase;
+
+/**
+ * DebVersionComparatorTest
+ */
+public class DebVersionComparatorTest extends TestCase {
+
+    private DebVersionComparator cmp;
+
+    protected void setUp() throws Exception {
+        super.setUp();
+        cmp = new DebVersionComparator();
+    }
+
+    protected void tearDown() throws Exception {
+        cmp = null;
+        super.tearDown();
+    }
+
+    public void testDpkgTestCases() {
```

Comment:
Could we re-use data in this method to also check the stored procedure implementation of the algorithm?

One way to do it could be to create a List of those versions and then sorting it with the Comparator, and inspecting the sorted result.

Reply:
This made me think and with your suggestions we moved the database tests into JUnit!
So we have now one single point to test the versions: Java (for Debian, and for RPM - as a bonus track - too!)